.gitignore

```diff
@@ -1,27 +1,127 @@
+# non-default terraform ignores
 *.aci
-.terraform
 .terraformrc
 config.tfvars
+terraform.tfvars
+
 id_rsa*
 pull-secret.json
 ssh-key.pem
 tectonic-*.tar.gz
 tectonic-license.txt
-terraform.tfstate*
-terraform.tfvars
+config.yaml
+
 bin/
 bin_test/
 matchbox/
 /contrib/govcloud/vpn.conf
 tectonic-dev
 
-# Bazel
+# non-default Bazel stuff
 .build/
-bazel-installer
-bazel-bin
-bazel-out
-bazel-genfiles
-bazel-testlogs
-bazel-tectonic-installer
-bazel-openshift-installer
 .cache
+
+# Created by https://www.gitignore.io/api/go,vim,bazel,emacs,terraform
```

Comment:
I'd rather drop the editors here.  From [`gitignore(5)`][1]:

> Patterns which a user wants Git to ignore in all situations (e.g., backup or temporary files generated by the user’s editor of choice) generally go into a file specified by `core.excludesFile` in the user’s `~/.gitconfig`. Its default value is `$XDG_CONFIG_HOME/git/ignore`. If `$XDG_CONFIG_HOME` is either not set or empty, `$HOME/.config/git/ignore` is used instead.

[1]: https://git-scm.com/docs/gitignore